scripts/gen-service-kustomize.sh

```diff
@@ -65,6 +65,14 @@ patches:
       path: /spec/storageClass
       value: ${STORAGE_CLASS}
 EOF
+if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}" != "" ]]; then
+    cat <<EOF >>kustomization.yaml
+    - op: add
+      path: /spec/neutron/template/customServiceConfig
+      value: |-
+$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
```

Comment:
I think this forces to keep the indentation of kustomization.yaml in the patch file provided. I believe this is a better option (the user should be agnostic about the format of the generated kustomization.yaml):
~~~
diff --git a/scripts/gen-service-kustomize.sh b/scripts/gen-service-kustomize.sh
index 5616334..3a50f7d 100644
--- a/scripts/gen-service-kustomize.sh
+++ b/scripts/gen-service-kustomize.sh
@@ -70,7 +70,7 @@ if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}"
     - op: add
       path: /spec/neutron/template/customServiceConfig
       value: |-
-$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
+      $(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
 EOF
 fi
~~~

Reply:
The above snippet only indent the first line of the file patch content. Something like this could valid:
~~~
diff --git a/scripts/gen-service-kustomize.sh b/scripts/gen-service-kustomize.sh
index 5616334..3d6da86 100644
--- a/scripts/gen-service-kustomize.sh
+++ b/scripts/gen-service-kustomize.sh
@@ -70,8 +70,11 @@ if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}"
     - op: add
       path: /spec/neutron/template/customServiceConfig
       value: |-
-$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
 EOF
+while IFS= read -r line
+do
+    echo "      $line" >>kustomization.yaml
+done < ${OPENSTACK_NEUTRON_CUSTOM_CONF}
 fi
 
 IFS=',' read -ra IMAGES <<< "$IMAGE"
~~~
Probably there are more elegant ways of doing this.

Reply:
> I think this forces to keep the indentation of kustomization.yaml in the patch file provided. I believe this is a better option (the user should be agnostic about the format of the generated kustomization.yaml):
> 
> ```
> diff --git a/scripts/gen-service-kustomize.sh b/scripts/gen-service-kustomize.sh
> index 5616334..3a50f7d 100644
> --- a/scripts/gen-service-kustomize.sh
> +++ b/scripts/gen-service-kustomize.sh
> @@ -70,7 +70,7 @@ if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}"
>      - op: add
>        path: /spec/neutron/template/customServiceConfig
>        value: |-
> -$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
> +      $(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
>  EOF
>  fi
> ```

Tried this also but no changes in kustomization.yaml.

Reply:
> The above snippet only indent the first line of the file patch content. Something like this could valid:
> 
> ```
> diff --git a/scripts/gen-service-kustomize.sh b/scripts/gen-service-kustomize.sh
> index 5616334..3d6da86 100644
> --- a/scripts/gen-service-kustomize.sh
> +++ b/scripts/gen-service-kustomize.sh
> @@ -70,8 +70,11 @@ if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}"
>      - op: add
>        path: /spec/neutron/template/customServiceConfig
>        value: |-
> -$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
>  EOF
> +while IFS= read -r line
> +do
> +    echo "      $line" >>kustomization.yaml
> +done < ${OPENSTACK_NEUTRON_CUSTOM_CONF}
>  fi
>  
>  IFS=',' read -ra IMAGES <<< "$IMAGE"
> ```
> 
> Probably there are more elegant ways of doing this.

I will place another PR to fix this alignment issue. will merge this PR, getting merge conflicts frequently.

Reply:
> > The above snippet only indent the first line of the file patch content. Something like this could valid:
> > ```
> > diff --git a/scripts/gen-service-kustomize.sh b/scripts/gen-service-kustomize.sh
> > index 5616334..3d6da86 100644
> > --- a/scripts/gen-service-kustomize.sh
> > +++ b/scripts/gen-service-kustomize.sh
> > @@ -70,8 +70,11 @@ if [[ "${KIND}" == "OpenStackControlPlane" && "${OPENSTACK_NEUTRON_CUSTOM_CONF}"
> >      - op: add
> >        path: /spec/neutron/template/customServiceConfig
> >        value: |-
> > -$(cat ${OPENSTACK_NEUTRON_CUSTOM_CONF})
> >  EOF
> > +while IFS= read -r line
> > +do
> > +    echo "      $line" >>kustomization.yaml
> > +done < ${OPENSTACK_NEUTRON_CUSTOM_CONF}
> >  fi
> >  
> >  IFS=',' read -ra IMAGES <<< "$IMAGE"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Probably there are more elegant ways of doing this.
> 
> I will place another PR to fix this alignment issue. will merge this PR, getting merge conflicts frequently.

but tested this PR and alignment is not a problem when using with value: |-